### PR TITLE
[Editor] Add support for printing/saving free highlight annotations

### PR DESCRIPTION
**Pull request by calixteman**

### CODE REVIEW

File: src/core/annotation.js [hunk 1]
```diff
@@ -4340,19 +4359,22 @@ class InkAnnotation extends MarkupAnnotation {
...
-    );
+    ink.set("InkList", outlines?.points || paths.map(p => p.points));
     ink.set("F", 4);
     ink.set("Rotate", rotation);
 
+    if (outlines) {
+      // Free highlight.
+      ink.set("IT", Name.get("InkHighlight"));
+    }
+
```
Comment (Snuffleupagus) [inline src/core/annotation.js:4377]: Looking at the PDF specification, I cannot find any mention of `IT` being supported for InkAnnotations; please see https://opensource.adobe.com/dc-acrobat-sdk-docs/pdfstandards/PDF32000_2008.pdf#page=400&zoom=auto,0,250:
> (Optional; PDF 1.6) A name describing the intent of the markup
annotation. Intents allow conforming readers to distinguish between
different uses and behaviors of a single markup annotation type. If this
entry is not present or its value is the same as the annotation type, the
annotation shall have no explicit intent and should behave in a generic
manner in a conforming reader.
>
> Free text annotations (Table 174), line annotations (Table 175), polygon
annotations (Table 178), and (PDF 1.7) polyline annotations (Table 178)
have defined intents, whose values are enumerated in the corresponding
tables.

Furthermore, searching for the string "InkHighlight" in the PDF specification produces exactly zero matches. What's the origin of this code, and should we actually set this when it doesn't seems to be defined in the specification?

If this is based on real-world behaviour, e.g. PDFs generated by Adobe software, then please add a comment to that effect to aid future readers!

File: src/core/writer.js [hunk 2]
```diff
@@ -32,7 +32,7 @@ async function writeObject(ref, obj, buffer, { encrypt = null }) {
     await writeDict(obj, buffer, transform);
   } else if (obj instanceof BaseStream) {
     await writeStream(obj, buffer, transform);
-  } else if (Array.isArray(obj)) {
+  } else if (Array.isArray(obj) || ArrayBuffer.isView(obj)) {
```
Comment (Snuffleupagus) [inline src/core/writer.js:35]: Here, and below, can't we use the existing [`isArrayBuffer` helper function](https://github.com/mozilla/pdf.js/blob/1cdbcfef821c7f6e81ea22fe68a8b815bca01c4e/src/shared/util.js#L966-L968) instead?

**Review by Snuffleupagus — CHANGES_REQUESTED** [review]
When saving a PDF document with a free-highlight, does the generated PDF hit [this work-around](https://github.com/mozilla/pdf.js/blob/1cdbcfef821c7f6e81ea22fe68a8b815bca01c4e/src/core/annotation.js#L4455-L4461) once opened with PDF.js?
Because if so we may need to fix that *properly* now, since otherwise we'd be ignoring the appearance-stream that we previously generated.

File: src/core/writer.js (hunk 2 above)
Comment (calixteman) [reply src/core/writer.js:35]: Why do we have this `isArrayBuffer` function ? Why not just using `ArrayBuffer.isView` ?

**Comment by calixteman** [comment]
> When saving a PDF document with a free-highlight, does the generated PDF hit [this work-around](https://github.com/mozilla/pdf.js/blob/1cdbcfef821c7f6e81ea22fe68a8b815bca01c4e/src/core/annotation.js#L4455-L4461) once opened with PDF.js? Because if so we may need to fix that _properly_ now, since otherwise we'd be ignoring the appearance-stream that we previously generated.

The free-highlight is a Ink annotation so the workaround doesn't apply here.

File: src/core/annotation.js (hunk 1 above)
Comment (calixteman) [reply src/core/annotation.js:4377]: I'll add a comment.
FYI, I found that when using the free highlighting in Edge and when opening the saved pdf in Acrobat  there is something indicating that the ink annotation has been used for highlighting (`surbrillance` in French):
image

**Review by Snuffleupagus — APPROVED** [review]
r=me, with passing tests; thank you!

**Pull request merged**